lib/trino-parquet/src/main/java/io/trino/parquet/ParquetReaderOptions.java

```diff
@@ -44,13 +47,15 @@ public ParquetReaderOptions()
     private ParquetReaderOptions(
             boolean ignoreStatistics,
             DataSize maxReadBlockSize,
+            int maxReadBlockRowCount,
             DataSize maxMergeDistance,
             DataSize maxBufferSize,
             boolean useColumnIndex,
             boolean useBatchColumnReaders)
     {
         this.ignoreStatistics = ignoreStatistics;
         this.maxReadBlockSize = requireNonNull(maxReadBlockSize, "maxReadBlockSize is null");
+        this.maxReadBlockRowCount = maxReadBlockRowCount;
```

Comment:
checkArgument that it's > 0

Reply:
Will also add such checks to the session properties

Reply:
I've also added such a check to the session properties